src/Authorization.AspNetCore/GraphQLBuilderAuthorizationExtensions.cs

```diff
@@ -33,5 +35,31 @@ public static IGraphQLBuilder AddGraphQLAuthorization(this IGraphQLBuilder build
 
             return builder;
         }
+
+        /// <summary>
+        /// Adds the GraphQL authorization.
+        /// </summary>
+        /// <param name="builder">The GraphQL builder.</param>
+        /// <returns>Reference to the passed <paramref name="builder"/>.</returns>
+        public static DI.IGraphQLBuilder AddGraphQLAuthorization(this DI.IGraphQLBuilder builder)
+            => builder.AddGraphQLAuthorization(_ => { });
+
+        /// <summary>
+        /// Adds the GraphQL authorization.
+        /// </summary>
+        /// <param name="builder">The GraphQL builder.</param>
+        /// <param name="configure">An action delegate to configure the provided <see cref="AuthorizationOptions"/>.</param>
+        /// <returns>Reference to the passed <paramref name="builder"/>.</returns>
+        public static DI.IGraphQLBuilder AddGraphQLAuthorization(this DI.IGraphQLBuilder builder, Action<AuthorizationOptions> configure)
+        {
+            if (!(builder is IServiceCollection services))
```

Comment:
Now I see...

Reply:
I would remove `: IServiceCollection` from `GraphQLBuilder` and add that:
```
public interface IServiceCollectionProvider
{
    public IServiceCollection ServiceCollection { get; }
}
```
Then
```
internal class GraphQLBuilder : GraphQLBuilderBase, IServiceRegister, IServiceCollectionProvider
{
        public override IServiceRegister Services => this;

        public IServiceCollection ServiceCollection { get; }
}
```
\+ remove 

```
 int ICollection<ServiceDescriptor>.Count => ServiceCollection.Count;
        bool ICollection<ServiceDescriptor>.IsReadOnly => ServiceCollection.IsReadOnly;
        ServiceDescriptor IList<ServiceDescriptor>.this[int index] { get => ServiceCollection[index]; set => ServiceCollection[index] = value; }
        int IList<ServiceDescriptor>.IndexOf(ServiceDescriptor item) => ServiceCollection.IndexOf(item);
        void IList<ServiceDescriptor>.Insert(int index, ServiceDescriptor item) => ServiceCollection.Insert(index, item);
        void IList<ServiceDescriptor>.RemoveAt(int index) => ServiceCollection.RemoveAt(index);
        void ICollection<ServiceDescriptor>.Add(ServiceDescriptor item) => ServiceCollection.Add(item);
        void ICollection<ServiceDescriptor>.Clear() => ServiceCollection.Clear();
        bool ICollection<ServiceDescriptor>.Contains(ServiceDescriptor item) => ServiceCollection.Contains(item);
        void ICollection<ServiceDescriptor>.CopyTo(ServiceDescriptor[] array, int arrayIndex) => ServiceCollection.CopyTo(array, arrayIndex);
        bool ICollection<ServiceDescriptor>.Remove(ServiceDescriptor item) => ServiceCollection.Remove(item);
        IEnumerator<ServiceDescriptor> IEnumerable<ServiceDescriptor>.GetEnumerator() => ServiceCollection.GetEnumerator();
        IEnumerator IEnumerable.GetEnumerator() => ((IEnumerable)ServiceCollection).GetEnumerator();
```

Then cast `builder.Services` to `IServiceCollectionProvider`

Reply:
That's fine (for v5 obviously).  This is the 3rd option I mentioned.  It will require a dependency on GraphQL.MicrosoftDI from the server project.  Since it only works for Microsoft DI, the dependency makes sense.

`GraphQLBuilder` should remain public so that if someone wants to add additional initialization code to `RegisterDefaultServices`, they don't need to rewrite the entire class.  There's no reason to make it internal/private.

Reply:
But... then project needs a reference to `IServiceCollectionProvider`... So... Let's just move `: IServiceCollection` from `IGraphQLBuilder` to `IServiceRegister`.

Reply:
If OK I'll do a PR to `graphql-net/develop`

Reply:
> GraphQLBuilder should remain public so that if someone wants to add additional initialization code to RegisterDefaultServices, they don't need to rewrite the entire class. There's no reason to make it internal/private.

OK.

>  It will require a dependency on GraphQL.MicrosoftDI from the server project. 

I think we will be able to go without it.